infrastructure/src/main/java/org/corfudb/infrastructure/DataStore.java

```diff
@@ -213,4 +232,35 @@ public synchronized void delete(@Nonnull String key,
     private String getKey(String prefix, String key) {
         return prefix + "_" + key;
     }
+
+    /**
+     * Cleanup the DataStore files of a specific type when the number
+     * of these files exceeds user-specified retention limit. Cleanup
+     * is always done on files with lower epochs.
+     */
+    private void dataStoreFileCleanup(String prefix, String key) {
+        if (!key.contains(prefix)) {
+            return;
+        }
+
+        File logDir = new File(logDirPath);
+        File[] foundFiles = logDir.listFiles((dir, name) -> name.startsWith(prefix));
+
+        if (foundFiles != null && foundFiles.length > dsFileRetention) {
+            Arrays.stream(foundFiles)
+                    .sorted(Comparator.comparingInt(file -> {
+                        // Extract epoch number from file name and cast to int for comparision
+                        Matcher matcher = Pattern.compile("\\d+").matcher(file.getName());
+                        return matcher.find(prefix.length()) ? Integer.parseInt(matcher.group()) : 0;
+                    }))
```

Comment:
log an info message along with all the epochs that have been deleted.

Reply:
Done in a slightly different way when actually deleting the file, since this part sorts and lists all the files including those don't need to be deleted.